controller/service.go

```diff
@@ -214,3 +257,37 @@ func (c *controller) allocateIPs(key string, svc *v1.Service) ([]net.IP, error)
 	// Okay, in that case just bruteforce across all pools.
 	return c.ips.Allocate(key, serviceIPFamily, k8salloc.Ports(svc), k8salloc.SharingKey(svc), k8salloc.BackendKey(svc))
 }
+
+func (c *controller) getDesiredLbIPs(svc *v1.Service) ([]net.IP, error) {
+	lbIPsStr := svc.Annotations[annotationLoadBalancerIPs]
+	if lbIPsStr == "" {
+		return nil, nil
+	}
+	var lbIPs []net.IP
+	desiredLbIPs := strings.Split(lbIPsStr, ",")
+	for _, desiredLbIP := range desiredLbIPs {
```

Comment:
We must ensure that if the ips are two, one is v4 and the other is v6.
Also, we must fail if there are more than 2.

Reply:
The `ipfamily.ForAddressesIPs` method call already does the check, so we may not need an explicit check.